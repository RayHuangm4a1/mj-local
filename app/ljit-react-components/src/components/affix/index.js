import React from 'react';
import { Affix as AntdAffix, } from 'antd';
import PropTypes from 'prop-types';
import cx from 'classnames';

const propTypes = {
	style: PropTypes.object,
	className: PropTypes.string,
	positionToTop: PropTypes.number,
	positionToBottom: PropTypes.number,
	children: PropTypes.any,
	onChange: PropTypes.func,
};
const defaultProps = {
	onChange: () => {},
};

function Affix(props) {
	const {
		style,
		className,
		positionToTop,
		positionToBottom,
		children,
		onChange,
	} = props;

	return (
		<AntdAffix
			style={style}
			className={cx('ljit-affix', className,)}
			offsetBottom={positionToBottom}
			offsetTop={positionToTop}
			onChange={onChange}
		>
			{children}
		</AntdAffix>
	);
}

Affix.propTypes = propTypes;
Affix.defaultProps = defaultProps;

export default Affix;
