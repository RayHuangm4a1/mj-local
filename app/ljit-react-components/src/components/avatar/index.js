import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import { Avatar as AntdAvatar, } from 'antd';
import Icon from '../icon';

const SizeEnums = {
	SMALL: 'small',
	DEFAULT: 'default',
};

const { SMALL, DEFAULT, } = SizeEnums;

const propTypes = {
	icon: PropTypes.string,
	style: PropTypes.object,
	size: PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.oneOf([SMALL, DEFAULT,]),
	]),
	src: PropTypes.string,
	alt: PropTypes.string,
	text: PropTypes.string,
	onError: PropTypes.func,
};
const defaultProps = {
};

class Avatar extends Component {
	render() {
		const { icon, style, size, src, alt, text, onError, } = this.props;
		const content = icon ? <Icon type={icon}></Icon> : text ;

		return (
			<AntdAvatar
				style={style}
				size={size}
				src={src}
				alt={alt}
				onError={onError}
			>
				{content}
			</AntdAvatar>
		);
	}
}

Avatar.propTypes = propTypes;
Avatar.defaultProps = defaultProps;
Avatar.SizeEnums = SizeEnums;

export default Avatar;
