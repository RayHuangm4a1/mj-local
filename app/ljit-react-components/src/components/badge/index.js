import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Badge as AntdBadge, } from 'antd';
import './style.styl';

const MAX_VISIBLE_COUNT = 99;
const PREFIX_CLASS = 'ljit-badge';

const propTypes = {
	className: PropTypes.string,
	count: PropTypes.node,
	overflowCount: PropTypes.number,
	isZeroVisible: PropTypes.bool,
	isBordered: PropTypes.bool,
	children: PropTypes.node,
};

const defaultProps = {
	className: '',
	overflowCount: MAX_VISIBLE_COUNT,
	isZeroVisible: false,
	isBordered: true,
};

const Badge = ({
	className,
	count,
	overflowCount,
	isZeroVisible,
	isBordered,
	children,
}) => (
	<AntdBadge
		className={cx(PREFIX_CLASS, {
			[`${PREFIX_CLASS}--bordered`]: isBordered,
			[`${PREFIX_CLASS}--no-bordered`]: !isBordered,
		}, className)}
		count={count}
		overflowCount={overflowCount}
		showZero={isZeroVisible}
	>
		{children}
	</AntdBadge>
);

Badge.propTypes = propTypes;
Badge.defaultProps = defaultProps;

export default Badge;
