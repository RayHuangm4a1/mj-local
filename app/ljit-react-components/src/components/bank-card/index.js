import React, { Component, } from 'react';
import InfoCard from '../info-card';
import LabelText from '../label-text';
import Button from '../button';
import { isDateValid, formatDate,  } from '../../lib/moment-utils';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './style.styl';
const PREFIX_CLASS = 'ljit-bank-card';

const propTypes = {
	dataSource: PropTypes.shape({
		bankName: PropTypes.string.isRequired,
		payer: PropTypes.string.isRequired,
		number: PropTypes.oneOfType([
			PropTypes.string,
			PropTypes.number,
		]).isRequired,
		activatedAt: PropTypes.string,
		withdrawableAt: PropTypes.string,
	}),
	isUnbindButtonVisible: PropTypes.bool,
	onClickUnbind: PropTypes.func,
	className: PropTypes.string,
};

const defaultProps = {
	isUnbindButtonVisible: false,
	onClickUnbind: () => {},
	className: '',
};

class BankCard extends Component {
	constructor() {
		super();

		this._renderUnbindButton = this._renderUnbindButton.bind(this);
		this._formatDate = this._formatDate.bind(this);
	}

	_renderUnbindButton() {
		const { dataSource, onClickUnbind, isUnbindButtonVisible, } = this.props;

		if (isUnbindButtonVisible) {
			return (
				<div className={`${PREFIX_CLASS}__bottom`}>
					<Button
						onClick={() => onClickUnbind(dataSource)}
						isFullWidth
					>
						解除绑定
					</Button>
				</div>
			);
		}
	}

	_formatDate(date) {
		if (isDateValid(date)) {
			return formatDate(date);
		}
		return '-';
	}

	render() {
		const { dataSource, className, } = this.props;
		const { _renderUnbindButton, _formatDate, } = this;
		const {
			bankName,
			payer,
			number,
			activatedAt,
			withdrawableAt,
		} = dataSource;

		return (
			<InfoCard
				className={cx(PREFIX_CLASS, className)}
				left={(
					<div className={`${PREFIX_CLASS}__left`}>
						<div className={`${PREFIX_CLASS}__title`}>{bankName}</div>
						<LabelText
							label="持卡人："
							text={payer}
							fontSize={LabelText.SizeEnums.SMALL}
						/>
						<LabelText
							label="尾号："
							text={number}
							fontSize={LabelText.SizeEnums.SMALL}
						/>
					</div>
				)}
				right={(
					<div className={`${PREFIX_CLASS}__right`}>
						<LabelText
							label="绑定时间"
							text={_formatDate(activatedAt)}
							fontSize={LabelText.SizeEnums.SMALL}
						/>
						<LabelText
							label="可提现时间："
							text={_formatDate(withdrawableAt)}
							fontSize={LabelText.SizeEnums.SMALL}
						/>
					</div>
				)}
				bottom={_renderUnbindButton()}
			/>
		);
	}
}

BankCard.propTypes = propTypes;
BankCard.defaultProps = defaultProps;

export default BankCard;
