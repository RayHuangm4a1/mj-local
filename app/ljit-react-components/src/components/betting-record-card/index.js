import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import TextButton from '../text-button';
import LabelText from '../label-text';
import InfoCard from '../info-card';
import StatusTag from '../status-tag';
import {
	TIME_SECONDS,
	formatDate,
	isDateValid,
} from '../../lib/moment-utils';
import './style.styl';

const TagStatusEnums = StatusTag.StatusEnums;
const PREFIX_CLASS = 'ljit-betting-info-card';
const StatusEnums = {
	NEW: 1,
	WIN: 2,
	LOSE: 3,
	DRAW: 4,
	FAILED: 5,
	CANCELED: 6,
	EXPIRED: 7,
	NOT_OPENED: 8,
	OPENING: 9,
};
const {
	NEW,
	WIN,
	LOSE,
	DRAW,
	FAILED,
	CANCELED,
	EXPIRED,
	NOT_OPENED,
	OPENING,
} = StatusEnums;
const StatusMap = {
	[OPENING]: { text: '开奖中', statusTag: TagStatusEnums.BET_OPENING, },
	[WIN]: { text: '中奖', statusTag: TagStatusEnums.BET_WIN, },
	[LOSE]: { text: '未中奖', statusTag: TagStatusEnums.BET_LOSE, },
	[NEW]: { text: '未开奖', statusTag: TagStatusEnums.BET_UNOPENED, },
	[NOT_OPENED]: { text: '未开奖', statusTag: TagStatusEnums.BET_UNOPENED, },
	[DRAW]: { text: '平局', statusTag: TagStatusEnums.BET_UNOPENED, },
	[CANCELED]: { text: '已撤单', statusTag: TagStatusEnums.BET_CANCELED, },
	[FAILED]: { text: '失败', statusTag: TagStatusEnums.BET_CANCELED, },
	[EXPIRED]: { text: '过期', statusTag: TagStatusEnums.BET_UNOPENED, },
};

const propTypes = {
	onClick: PropTypes.func,
	onClickCancel: PropTypes.func,
	bettingRecord: PropTypes.shape({
		id: PropTypes.number,
		name: PropTypes.string,
		lotteryId: PropTypes.number,
		lotteryName: PropTypes.string,
		playId: PropTypes.number,
		unit: PropTypes.number,
		amountPerBet: PropTypes.number,
		amount: PropTypes.number,
		multiple: PropTypes.number,
		reward: PropTypes.number,
		rebate: PropTypes.number,
		issue: PropTypes.string,
		opencode: PropTypes.string,
		count: PropTypes.number,
		betcontent: PropTypes.string,
		weizhi: PropTypes.string,
		device: PropTypes.string,
		isPK: PropTypes.bool,
		details: PropTypes.arrayOf(PropTypes.shape({
			name: PropTypes.string,
			count: PropTypes.number,
			reward: PropTypes.number,
		})),
		status:  PropTypes.oneOf([
			NEW,
			WIN,
			LOSE,
			DRAW,
			FAILED,
			CANCELED,
			EXPIRED,
			NOT_OPENED,
			OPENING,
		]),
		createdAt: PropTypes.instanceOf(Date),
	}),
	className: PropTypes.string,
};
const defaultProps = {
	onClick: () => {},
	onClickCancel: () => {},
	bettingRecord: {},
	className: '',
};

class BettingRecordCard extends Component {
	constructor() {
		super();

		this._handleClickCancel = this._handleClickCancel.bind(this);
		this._renderCancelButton = this._renderCancelButton.bind(this);
	}

	_handleClickCancel(e) {
		const { onClickCancel, } = this.props;

		e.stopPropagation();
		onClickCancel();
	}
	_renderCancelButton() {
		const { _handleClickCancel, } = this;

		return (
			<TextButton
				text="撤單"
				onClick={_handleClickCancel}
				className={`${PREFIX_CLASS}__cancel`}
				fontSize={TextButton.SizeEnums.SMALL}
				fontWeight={TextButton.WeightEnums.HEAVY}
			/>
		);
	}
	render() {
		const { bettingRecord, className, onClick, } = this.props;
		const { _renderCancelButton, } = this;
		const {
			id,
			lotteryName,
			name,
			amountPerBet,
			reward,
			issue,
			createdAt,
			multiple,
			opencode,
			status = NOT_OPENED,
		} = bettingRecord;
		const isShowingCancelButton = (status === NOT_OPENED);

		return (
			<div
				className={cx(PREFIX_CLASS, className)}
				onClick={onClick}
			>
				<InfoCard
					className={`${PREFIX_CLASS}__info`}
					topLeft={(
						<div className={`${PREFIX_CLASS}__header`}>
							<span className={`${PREFIX_CLASS}__title`}>{lotteryName}</span>
							{isShowingCancelButton ? _renderCancelButton() : null}
							{StatusMap[status] ? <StatusTag status={StatusMap[status].statusTag} text={StatusMap[status].text}/> : null}
						</div>
					)}
					left={(
						<React.Fragment>
							<LabelText
								label="方案号"
								text={id}
								labelColType={LabelText.SizeEnums.MEDIUM}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="玩法"
								text={name}
								labelColType={LabelText.SizeEnums.MEDIUM}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="单注金额"
								text={amountPerBet}
								labelColType={LabelText.SizeEnums.LARGE}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="中奖金额"
								text={reward ? reward : '-'}
								labelColType={LabelText.SizeEnums.LARGE}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
						</React.Fragment>
					)}
					right={(
						<React.Fragment>
							<LabelText
								label="期号"
								text={issue}
								labelColType={LabelText.SizeEnums.SMALL}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="时间"
								text={isDateValid(createdAt) ? formatDate(createdAt, `M/D ${TIME_SECONDS}`) : '-'}
								labelColType={LabelText.SizeEnums.SMALL}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="倍数"
								text={multiple}
								labelColType={LabelText.SizeEnums.SMALL}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
							<LabelText
								label="中奖号码"
								text={opencode ? opencode : '-'}
								labelColType={LabelText.SizeEnums.LARGE}
								fontSize={LabelText.SizeEnums.SMALL}
							/>
						</React.Fragment>
					)}
				/>
			</div>
		);
	}
}

BettingRecordCard.propTypes = propTypes;
BettingRecordCard.defaultProps = defaultProps;

BettingRecordCard.StatusEnums = StatusEnums;
BettingRecordCard.StatusMap = StatusMap;

export default BettingRecordCard;
