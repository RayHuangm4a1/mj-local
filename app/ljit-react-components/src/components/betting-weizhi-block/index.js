import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import CheckBox from '../checkbox';
import Icon from '../icon';
import './style.styl';

const CLASS_PREFIX = 'ljit-betting-weizhi-block';

const CheckboxStyleEnum = {
	SOLID: 'solid',
	HOLLOW: 'hollow',
};

const {
	SOLID,
	HOLLOW,
} = CheckboxStyleEnum;

const propTypes = {
	className: PropTypes.string,
	options: PropTypes.arrayOf(PropTypes.shape({
		name: PropTypes.string,
		isSelected: PropTypes.bool,
	})).isRequired,
	onPressCheckbox: PropTypes.func,
	description: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.element,
	]),
	checkboxStyle: PropTypes.oneOf([
		SOLID,
		HOLLOW,
	]),
};

const defaultProps = {
	options: [],
	description: '',
	onPressCheckbox: () => {},
	checkboxStyle: SOLID,
};

class BettingWeizhiBlock extends Component {
	constructor() {
		super();

		this._renderOptions = this._renderOptions.bind(this);
		this._renderDescription = this._renderDescription.bind(this);
	}
	_renderOptions() {
		const { options, onPressCheckbox, } = this.props;
		const checkBoxes = options.map((option, index) => {
			return (
				<CheckBox
					key={option.name}
					value={option.isSelected}
					onChange={() => onPressCheckbox(index)}
				>
					{option.name}
				</CheckBox>
			);
		});

		return checkBoxes;
	}
	_renderDescription() {
		const { description, } = this.props;

		return (
			<div className={`${CLASS_PREFIX}__description`}>
				<Icon 
					type={Icon.IconTypeEnums.INFO_FILL}
					color={Icon.ColorEnums.PRIMARY}
					size={Icon.SizeEnums.SMALL}
				/>
				<span>
					{description}
				</span>
			</div>
		);
	}
	render() {
		const {
			className,
			description,
			checkboxStyle,
		} = this.props;
		const {
			_renderOptions,
			_renderDescription,
		} = this;

		return (
			<div className={cx(CLASS_PREFIX, {
				[`${CLASS_PREFIX}--solid`]: checkboxStyle === SOLID,
				[`${CLASS_PREFIX}--hollow`]: checkboxStyle === HOLLOW,
			}, className)}>
				{_renderOptions()}
				{description ? _renderDescription() : null}
			</div>
		);
	}
}

BettingWeizhiBlock.propTypes = propTypes;
BettingWeizhiBlock.defaultProps = defaultProps;

BettingWeizhiBlock.CheckboxStyleEnum = CheckboxStyleEnum;

export default BettingWeizhiBlock;
