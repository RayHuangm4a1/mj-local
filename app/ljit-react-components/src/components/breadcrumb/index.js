import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumbs, BreadcrumbsProvider, } from 'react-breadcrumbs-dynamic';
import cx from 'classnames';
import BreadcrumbItem from './item';
import './style.styl';

export const prefixClass = 'ljit-breadcrumb';

const propTypes = {
	className: PropTypes.string,
	Item: PropTypes.element,
};

const Breadcrumb = ({
	className,
	Item = 'a',
}) => {
	const itemClass = `${prefixClass}__item`;

	return (
		<Breadcrumbs
			container={(containerProps) => (
				<div
					{...containerProps}
					className={cx(prefixClass, className)}
				/>
			)}
			item={(itemProps) => {
				if (itemProps['data-isactive'] === true) {
					return <Item {...itemProps} />;
				}
				return React.createElement('span', itemProps);
			}}
			separator={<span className={`${prefixClass}__separator`}>/</span>}
			finalItem={'span'}
			finalProps={{
				className: `${itemClass} ${itemClass}--inactive`,
			}}
		/>
	);
};

Breadcrumb.propTypes = propTypes;

Breadcrumb.BreadcrumbProvider = BreadcrumbsProvider;
Breadcrumb.BreadcrumbItem = BreadcrumbItem;

export default Breadcrumb;
