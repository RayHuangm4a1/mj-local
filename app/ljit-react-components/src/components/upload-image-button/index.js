import React, { Component, } from 'react';
import PropTypes from 'prop-types';
import {
	Icon, Button,
} from 'antd';
import {
	ManualUpload,
} from '../../';
import cx from 'classnames';

export const AcceptExtentions = ['gif', 'jpg', 'jpeg', 'png', 'bmp',];

const remindAccept = AcceptExtentions.join('、');

const propTypes = {
	text: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node,
	]),
	remindText: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node,
	]),
	onChange: PropTypes.func.isRequired,
	className: PropTypes.string,
	fileList: PropTypes.array,
	fileLimit: PropTypes.number,
};
const defaultProps = {
	text: '选择档案',
	remindText: `允许的副档名：${remindAccept}。`,
	fileLimit: 1,
};

class UploadImageButton extends Component {
	constructor() {
		super();
		this._handleFileChange = this._handleFileChange.bind(this);
		this._handelFileBeforeUpload = this._handelFileBeforeUpload.bind(this);
	}

	_handleFileChange(info) {
		this.props.onChange(info.fileList);
	}

	_handelFileBeforeUpload(file) {
		this.props.onChange([...this.props.fileList, file,]);
	}

	render() {
		const {
			text,
			remindText,
			className,
			fileList,
			fileLimit,
		} = this.props;

		return (
			<ManualUpload
				className={cx('ljit-upload-image-button', className)}
				acceptExtentions={AcceptExtentions}
				listType={ManualUpload.ListTypeEnums.PICTURE}
				fileList={fileList}
				onChange={this._handleFileChange}
				beforeUpload={this._handelFileBeforeUpload}
				remindText={remindText}
				fileLimit={fileLimit}
			>
				<Button>
					<Icon type="upload" />{text}
				</Button>
			</ManualUpload>
		);
	}
}

UploadImageButton.propTypes = propTypes;
UploadImageButton.defaultProps = defaultProps;

export default UploadImageButton;
