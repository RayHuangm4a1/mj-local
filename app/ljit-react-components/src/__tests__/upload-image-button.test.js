import React from 'react';
import { shallow, mount, } from 'enzyme';
import { AcceptExtentions, } from '../components/upload-image-button';

describe('Upload Image Button', () => {
	let UploadImageButton;

	beforeEach(() => {
		jest.doMock('antd/lib/upload');
		UploadImageButton = require('../components/upload-image-button').default;
	});
	afterEach(() => {
		jest.unmock('antd/lib/upload');
		UploadImageButton = undefined;
	});

	it('handle default accept', () => {
		expect(AcceptExtentions).toMatchObject(['gif', 'jpg', 'jpeg', 'png', 'bmp',]);
	});

	it('handle default props', () => {
		const {
			text,
			remindText,
			fileLimit,
		} = UploadImageButton.defaultProps;

		expect(text).toBe('选择档案');
		expect(remindText).toBe('允许的副档名：gif、jpg、jpeg、png、bmp。');
		expect(fileLimit).toBe(1);
	});

	it('should renders correctly', () => {
		const wrapper = shallow(<UploadImageButton onChange={() => {}} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const text = 'mock-text';
		const remindText = 'mock-remindtext';
		const onChange = () => {};
		const className = 'mock-calss';
		const fileList = [];
		const fileLimit = 1;
		const wrapper = mount(
			<UploadImageButton
				text={text}
				remindText={remindText}
				onChange={onChange}
				className={className}
				fileList={fileList}
				fileLimit={fileLimit}
			/>
		);

		expect(wrapper.props().text).toBe(text);
		expect(wrapper.props().remindText).toBe(remindText);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().fileList).toEqual(fileList);
		expect(wrapper.props().fileLimit).toBe(fileLimit);
	});
});
