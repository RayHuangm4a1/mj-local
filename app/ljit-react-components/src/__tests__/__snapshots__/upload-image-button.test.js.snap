// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Image Button should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadImageButton
    fileLimit={1}
    onChange={[Function]}
    remindText="允许的副档名：gif、jpg、jpeg、png、bmp。"
    text="选择档案"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "acceptExtentions": Array [
        "gif",
        "jpg",
        "jpeg",
        "png",
        "bmp",
      ],
      "beforeUpload": [Function],
      "children": <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
      >
        <Icon
          type="upload"
        />
        选择档案
      </Button>,
      "className": "ljit-upload-image-button",
      "fileLimit": 1,
      "fileList": undefined,
      "isUploadListVisible": true,
      "listType": "picture",
      "onChange": [Function],
      "onRemove": [Function],
      "remindText": "允许的副档名：gif、jpg、jpeg、png、bmp。",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "block": false,
        "children": Array [
          <Icon
            type="upload"
          />,
          "选择档案",
        ],
        "ghost": false,
        "htmlType": "button",
        "loading": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "upload",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "选择档案",
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "acceptExtentions": Array [
          "gif",
          "jpg",
          "jpeg",
          "png",
          "bmp",
        ],
        "beforeUpload": [Function],
        "children": <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
        >
          <Icon
            type="upload"
          />
          选择档案
        </Button>,
        "className": "ljit-upload-image-button",
        "fileLimit": 1,
        "fileList": undefined,
        "isUploadListVisible": true,
        "listType": "picture",
        "onChange": [Function],
        "onRemove": [Function],
        "remindText": "允许的副档名：gif、jpg、jpeg、png、bmp。",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": false,
          "children": Array [
            <Icon
              type="upload"
            />,
            "选择档案",
          ],
          "ghost": false,
          "htmlType": "button",
          "loading": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "upload",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "选择档案",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
